exchanges/btcmarkets/btcmarkets_wrapper.go

```diff
@@ -272,60 +288,53 @@ func (b *BTCMarkets) FetchOrderbook(p currency.Pair, assetType asset.Item) (orde
 // UpdateOrderbook updates and returns the orderbook for a currency pair
 func (b *BTCMarkets) UpdateOrderbook(p currency.Pair, assetType asset.Item) (orderbook.Base, error) {
 	var orderBook orderbook.Base
-	orderbookNew, err := b.GetOrderbook(p.Base.String(),
-		p.Quote.String())
+	tempResp, err := b.GetOrderbook(b.FormatExchangeCurrency(p, assetType).String(), 2)
 	if err != nil {
 		return orderBook, err
 	}
 
-	for x := range orderbookNew.Bids {
-		data := orderbookNew.Bids[x]
-		orderBook.Bids = append(orderBook.Bids, orderbook.Item{Amount: data[1], Price: data[0]})
-	}
-
-	for x := range orderbookNew.Asks {
-		data := orderbookNew.Asks[x]
-		orderBook.Asks = append(orderBook.Asks, orderbook.Item{Amount: data[1], Price: data[0]})
+	for x := range tempResp.Bids {
+		orderBook.Bids = append(orderBook.Bids, orderbook.Item{
+			Amount: tempResp.Bids[x].Volume,
+			Price:  tempResp.Bids[x].Price})
 	}
 
-	orderBook.Pair = p
-	orderBook.ExchangeName = b.GetName()
-	orderBook.AssetType = assetType
+	for y := range tempResp.Asks {
+		orderBook.Asks = append(orderBook.Asks, orderbook.Item{
+			Amount: tempResp.Asks[y].Volume,
+			Price:  tempResp.Asks[y].Price})
+		orderBook.Pair = p
```

Comment:
This and below assignments should be outside of the for loop, otherwise orderbook.Process will be called every iteration